app/src/main/java/com/m2049r/xmrwallet/WalletFragment.java

```diff
@@ -374,6 +374,10 @@ public void unsync() {
             bSend.setEnabled(false);
         }
         if (isVisible()) enableAccountsList(false); //otherwise it is enabled in onResume()
+    }
+
+    public void onStartRescan() {
+        unsync();
         firstBlock = 0;
```

Comment:
i am not sure not unsetting `firstBlock` in `unsync()` has no adverse effects

Reply:
Is there another place `unsync()` is called I'm not seeing except in the place I added?

Reply:
Will dig into it/test it some more. The issue I was thinking with resetting it in that new call to `unsync` I added is if it might cause continual resets of the progress bar

Reply:
i don*t see another place either. yes, i mean also unsetting `firstBlock` - maybe even scrapping this method as its only used once and has no obvious reuse scenario (or?).